keel-api/src/main/kotlin/com/netflix/spinnaker/keel/rest/DeliveryConfigController.kt

```diff
@@ -45,5 +55,50 @@ class DeliveryConfigController(
   fun diff(@RequestBody deliveryConfig: SubmittedDeliveryConfig): List<EnvironmentDiff> =
     adHocDiffer.calculate(deliveryConfig)
 
+  @GetMapping(
+    path = ["/{name}/environment/{environment}/constraint"],
+    produces = [APPLICATION_JSON_VALUE, APPLICATION_YAML_VALUE]
+  )
+  fun getConstraintState(
+    @PathVariable("name") name: String,
+    @PathVariable("environment") environment: String,
+    @RequestParam("limit") limit: Int?
+  ): List<ConstraintState> =
+    deliveryConfigRepository.constraintStateFor(name, environment, limit ?: DEFAULT_MAX_EVENTS)
+
+  @PostMapping(
+    path = ["/{name}/environment/{environment}/constraint/{type}"],
+    consumes = [APPLICATION_JSON_VALUE, APPLICATION_YAML_VALUE],
+    produces = [APPLICATION_JSON_VALUE, APPLICATION_YAML_VALUE]
+  )
+  // TODO: This should be validated against write access to a service account. Service accounts should
+  //  become a top-level property of either delivery configs or environments.
+  fun updateConstraintStatus(
+    @RequestHeader("X-SPINNAKER-USER") user: String,
+    @PathVariable("name") name: String,
+    @PathVariable("environment") environment: String,
+    @PathVariable("type") type: String,
+    @RequestParam("artifactVersion") artifactVersion: String,
+    @RequestParam("status") status: String,
+    @RequestParam("comment") comment: String?
+  ) {
+    val currentState = deliveryConfigRepository.getConstraintState(name, environment, artifactVersion, type)
+      ?: throw InvalidConstraintException("$name/$environment/$type/$artifactVersion", "constraint not found")
+
+    deliveryConfigRepository.storeConstraintState(
+      currentState.copy(
+        status = ConstraintStatus.valueOf(status),
+        comment = comment ?: currentState.comment,
+        judgedAt = Instant.now(),
+        judgedBy = user
```

Comment:
I think I missed this before, but isn't this specific to the `manual-judgement` constraint type?

Reply:
No, these are intended to apply to all stateful constraints.

Reply:
Hmm... Can you please give me an example of judgement on a stateful constraint besides manual judgement? Maybe it's just the naming of these properties that is throwing me off.

Reply:
Canary was the example discussed in the original PR.

Reply:
1) Constraint evaluators always produce a boolean judgement. 
2) Automated judgements can always be overridden by an operator.
3) Automated constraint types shouldn't necessarily be tied to an implementation. A `canary` constraint type that is judged by "Kayenta" today may have been judged by "ACA" yesterday without requiring changes to a service owner's delivery config.

Reply:
Makes sense. I think for me it was just the term overlap with `manual-judgement` that made it sound specific for me (as opposed to "approval" or "promotion" as used in other constraint-related interfaces). Maybe some room for renaming, but I can't come up with a better name at the moment.